src/actors/symbolication.rs

```diff
@@ -1011,53 +1011,24 @@ pub struct SymbolicateStacktraces {
     pub options: RequestOptions,
 }
 
-/// Run future with a timeout, instrumented with metrics.
-///
-/// This polls the [`Future`] to completion and exposes the duration as the `"futures.done"` metric
-/// using `"task_name"` as metric tag. The future will be cancelled if it runs for longer than
-/// `duration` with this status also being represented in the metric.
-async fn measure_task_timeout<T, F>(task_name: &str, future: F, duration: Duration) -> F::Output
-where
-    F: Future<Output = Result<T, SymbolicationError>>,
-{
-    let start_time = Instant::now();
-    let delay = Delay::new(start_time + duration);
-
-    let res = select! {
-        output = future.fuse() => output,
-        _ = delay.compat().fuse() => Err(SymbolicationError::Timeout),
-    };
-    let status = match res {
-        Ok(_) => "ok",
-        Err(SymbolicationError::Timeout) => "timeout",
-        Err(_) => "err",
-    };
-    metric!(
-        timer("futures.done") = start_time.elapsed(),
-        "task_name" => task_name,
-        "status" => status,
-    );
-    res
-}
-
 impl SymbolicationActor {
     async fn do_symbolicate(
         self,
         request: SymbolicateStacktraces,
     ) -> Result<CompletedSymbolicationResponse, SymbolicationError> {
         let serialize_dif_candidates = request.options.dif_candidates;
-        measure_task_timeout(
-            "symbolicate",
-            self.do_symbolicate_impl(request),
-            Duration::from_secs(3600),
-        )
-        .await
-        .map(|mut response| {
-            if !serialize_dif_candidates {
-                response.clear_dif_candidates();
-            }
-            response
-        })
+
+        let f = self.do_symbolicate_impl(request);
+        let f = timeout_compat(Duration::from_secs(3600), f);
+        let f = measure("symbolicate", m::timed_result, f);
```

Comment:
This emits a new metric, `futures.wait_time`.  And since this is not being spawned it will always be super-short since it's being polled right away.  This will seriously skew our `futures.wait_time` metric to the point it's not useful as a metric to look at.
Luckily our default dashboard groups by task_name already, but I still don't think this is nice.  Can we have a variation which doesn't measure `futures.wait_time`?  Operationally having to know when this metric tells something meaningful and when not is not nice, it's mere presence indicates it's showing spawn-time and this violates that.

Reply:
The `futures.wait_time` metric did exist and was only dropped in a recent refactor of how we report future metrics since we said it was not super crucial to us. It was never meant to be read across tasks, so nothing changes in the contract.

Reply:
doesn't change the reasoning why this is a bad idea: `futures.wait_time` measures the time a future was waiting before it first gets polled, to add it to futures which are already being polled makes no sense and gives a false impression when looking at the metrics.